doc/sphinxext/gallery_generator.py

```diff
@@ -59,136 +59,33 @@
-        -o-filter: blur(3px);
-        -ms-filter: blur(3px);
-        filter: blur(3px);
-        opacity:1.0;
-        filter:alpha(opacity=100); /* For IE8 and earlier */
-    }}
-
-    span.figure-label {{
-        position: absolute;
-        display: inline;
-        left: 0;
-        width: 170px;
-        height: 170px;
-        background: #000;
-        color: #fff;
-        visibility: hidden;
-        opacity: 0;
-        z-index: 100;
-    }}
-
-    .figure p {{
-        position: absolute;
-        top: 45%;
-        width: 170px;
-        font-size: 110%;
-    }}
-
-    .figure:hover span {{
-        visibility: visible;
-        opacity: .4;
-    }}
-
-    .caption {{
-        position: absolute;
-        width: 180px;
-        top: 170px;
-        text-align: center !important;
-    }}
-
-    .figure .gallery-figure-title p {{
-        position: relative;
-        top: 170px;
-        color: black;
-        visibility: visible;
-        text-align: center !important;
-        line-height: normal;
-    }}
-    .figure .gallery-figure-title span {{
-        top: 170px;
-        position: relative;
-        visibility: visible;
-    }}
-    </style>
-
-.. _{sphinx_tag}:
-
 Example gallery
 ===============
+{toctree}
+{contents}
+"""
 
-{toctrees_contents}
+TOCTREE_START = """
+.. toctree::
+   :hidden:
 
 """
 
-CONTENTS_ENTRY_TEMPLATE = (
-    ".. raw:: html\n\n"
-    "    <div class='figure align-center'>\n"
-    "    <a href=./{htmlfilename}>\n"
-    "    <img src=../_static/{thumbfilename}>\n"
-    "    <span class='figure-label'>\n"
-    "    <p>{sphinx_tag}</p>\n"
-    "    </span>\n"
-    '    <span class="gallery-figure-title">\n'
-    "      <p>{title}</p>\n"
-    "    </span>\n"
-    "    </a>\n"
-    "    </div>\n\n"
-    "\n\n"
-    ""
-)
-
-
-def create_thumbnail(infile, thumbfile, width=275, height=275, cx=0.5, cy=0.5, border=4):
-
-    im = image.imread(infile)
-    rows, cols = im.shape[:2]
-    size = min(rows, cols)
-    if size == cols:
-        xslice = slice(0, size)
-        ymin = min(max(0, int(cy * rows - size // 2)), rows - size)
-        yslice = slice(ymin, ymin + size)
-    else:
-        yslice = slice(0, size)
-        xmin = min(max(0, int(cx * cols - size // 2)), cols - size)
-        xslice = slice(xmin, xmin + size)
-    thumb = im[yslice, xslice]
-    thumb[:border, :, :3] = thumb[-border:, :, :3] = 0
-    thumb[:, :border, :3] = thumb[:, -border:, :3] = 0
-
-    dpi = 100
-    fig = plt.figure(figsize=(width / dpi, height / dpi), dpi=dpi, constrained_layout=True)
-
-    ax = fig.add_axes([0, 0, 1, 1], aspect="auto", frameon=False, xticks=[], yticks=[])
-    ax.imshow(thumb, aspect="auto", resample=True, interpolation="bilinear")
-    fig.savefig(thumbfile, dpi=dpi)
-    plt.close(fig)
+CONTENTS_START = """
+.. grid:: 3
+   :gutter: 2 2 3 3
+"""
+
+CONTENTS_ENTRY_TEMPLATE = """
+   .. grid-item-card:: 
+      :link: ./{htmlfilename}
+      :img-top: ./matplotlib/_images/{imgfilename}
```

Comment:
This confused me a lot actually... I sort of guessed until it magically worked 😂 

If you open up the individual RST file, the image path is relative to that file. For example, `source/examples/index.rst` would need the image source as `./matplotlib/_images/*.png`. It will not match with the rendered page, which dumps all the images in `html/_images`.

Reply:
that's true, sphinx moves all the images to `_images/` folder during the build process, independently of their original relative path inisde the source files, only the paths of the source files are preserved into the urls. Thanks for the reminder